--- scipy-0.13.3/scipy/cluster/setup.py.lm~	2014-02-28 06:46:05.158989774 +0100
+++ scipy-0.13.3/scipy/cluster/setup.py	2014-02-28 06:50:32.218994987 +0100
@@ -20,11 +20,13 @@ def configuration(parent_package='', top
     config.add_extension('_vq',
         sources=[join('src', 'vq_module.c'), join('src', 'vq.c')],
         include_dirs=[get_numpy_include_dirs()],
+        libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     config.add_extension('_hierarchy_wrap',
         sources=[join('src', 'hierarchy_wrap.c'), join('src', 'hierarchy.c')],
         include_dirs=[get_numpy_include_dirs()],
+        libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     return config
--- scipy-0.13.3/scipy/interpolate/setup.py.lm~	2014-02-04 05:56:48.000000000 +0100
+++ scipy-0.13.3/scipy/interpolate/setup.py	2014-02-28 06:46:05.158989774 +0100
@@ -13,7 +13,8 @@ def configuration(parent_package='',top_
     config.add_library('fitpack', sources=fitpack_src)
 
     config.add_extension('interpnd',
-                         sources=['interpnd.c'])
+                         sources=['interpnd.c'],
+    			 libraries=['m'])
 
     config.add_extension('_fitpack',
                          sources=['src/_fitpackmodule.c'],
@@ -29,6 +30,7 @@ def configuration(parent_package='',top_
                          )
 
     config.add_extension('_interpolate',
+			 libraries = ['m'],
                          sources=['src/_interpolate.cpp'],
                          include_dirs=['src'],
                          depends=['src/interpolate.h'])
--- scipy-0.13.3/scipy/ndimage/setup.py.lm~	2014-02-03 14:48:50.000000000 +0100
+++ scipy-0.13.3/scipy/ndimage/setup.py	2014-02-28 06:46:05.158989774 +0100
@@ -14,6 +14,7 @@ def configuration(parent_package='', top
                  "src/ni_fourier.c","src/ni_interpolation.c",
                  "src/ni_measure.c",
                  "src/ni_morphology.c","src/ni_support.c"],
+	libraries = ['m'],
         include_dirs=['src']+[get_include()],
     )
 
--- scipy-0.13.3/scipy/optimize/setup.py.lm~	2014-02-03 14:48:50.000000000 +0100
+++ scipy-0.13.3/scipy/optimize/setup.py	2014-02-28 06:51:39.718996305 +0100
@@ -21,7 +21,8 @@ def configuration(parent_package='',top_
     rootfind_hdr = [join('Zeros','zeros.h')]
     config.add_library('rootfind',
                        sources=rootfind_src,
-                       headers=rootfind_hdr)
+                       headers=rootfind_hdr,
+    		       libraries=['m'])
 
     config.add_extension('_zeros',
                          sources=['zeros.c'],
@@ -37,7 +38,8 @@ def configuration(parent_package='',top_
     sources = ['moduleTNC.c','tnc.c']
     config.add_extension('moduleTNC',
                          sources=[join('tnc',x) for x in sources],
-                         depends=[join('tnc','tnc.h')])
+                         depends=[join('tnc','tnc.h')],
+			 libraries=['m'])
 
     config.add_extension('_cobyla',
                          sources=[join('cobyla',x) for x in ['cobyla.pyf',
--- scipy-0.13.3/scipy/signal/setup.py.lm~	2014-02-28 06:46:05.158989774 +0100
+++ scipy-0.13.3/scipy/signal/setup.py	2014-02-28 06:53:14.328998152 +0100
@@ -14,14 +14,16 @@ def configuration(parent_package='', top
                                   'medianfilter.c', 'lfilter.c.src',
                                   'correlate_nd.c.src'],
                          depends=['sigtools.h'],
-                         include_dirs=['.']
+                         include_dirs=['.'],
+                         libraries = ['m']
                          )
 
-    config.add_extension('_spectral', sources=['_spectral.c'])
+    config.add_extension('_spectral', sources=['_spectral.c'], libraries = ['m'])
 
     config.add_extension('spline',
         sources=['splinemodule.c', 'S_bspline_util.c', 'D_bspline_util.c',
                  'C_bspline_util.c', 'Z_bspline_util.c', 'bspline_util.c'],
+        libraries = ['m']
     )
 
     return config
--- scipy-0.13.3/scipy/spatial/setup.py.lm~	2014-02-28 06:46:05.158989774 +0100
+++ scipy-0.13.3/scipy/spatial/setup.py	2014-02-28 06:54:11.918999276 +0100
@@ -33,10 +33,11 @@ def configuration(parent_package='', top
                          sources=['qhull.c'] + qhull_src,
                          **cfg)
 
-    config.add_extension('ckdtree', sources=['ckdtree.c'])  # FIXME: cython
+    config.add_extension('ckdtree', sources=['ckdtree.c'], libraries = ['m'])  # FIXME: cython
 
     config.add_extension('_distance_wrap',
         sources=[join('src', 'distance_wrap.c'), join('src', 'distance.c')],
+        libraries = ['m'],
         include_dirs=[get_numpy_include_dirs()])
 
     return config
--- scipy-0.13.3/scipy/stats/setup.py.lm~	2014-02-03 14:48:50.000000000 +0100
+++ scipy-0.13.3/scipy/stats/setup.py	2014-02-28 06:46:05.158989774 +0100
@@ -23,6 +23,7 @@ def configuration(parent_package='',top_
 
     # add vonmises_cython module
     config.add_extension('vonmises_cython',
+	libraries = ['m'],
         sources=['vonmises_cython.c'],  # FIXME: use cython source
     )
 
