--- scipy-0.12.0/scipy/cluster/setup.py.orig	2013-07-20 21:11:49.493279989 -0300
+++ scipy-0.12.0/scipy/cluster/setup.py	2013-07-20 21:13:57.824285848 -0300
@@ -19,11 +19,13 @@ def configuration(parent_package = '', t
     config.add_extension('_vq',
         sources=[join('src', 'vq_module.c'), join('src', 'vq.c')],
         include_dirs = [get_numpy_include_dirs()],
+	libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     config.add_extension('_hierarchy_wrap',
         sources=[join('src', 'hierarchy_wrap.c'), join('src', 'hierarchy.c')],
         include_dirs = [get_numpy_include_dirs()],
+	libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     return config
--- scipy-0.12.0/scipy/interpolate/setup.py.orig	2013-07-20 21:11:59.953280467 -0300
+++ scipy-0.12.0/scipy/interpolate/setup.py	2013-07-20 21:14:38.362287698 -0300
@@ -13,7 +13,8 @@ def configuration(parent_package='',top_
                       )
 
     config.add_extension('interpnd',
-                         sources=['interpnd.c'])
+                         sources=['interpnd.c'],
+    			 libraries=['m'])
 
     config.add_extension('_fitpack',
                          sources=['src/_fitpackmodule.c'],
@@ -29,6 +30,7 @@ def configuration(parent_package='',top_
     config.add_extension('_interpolate',
                          sources=['src/_interpolate.cpp'],
                          include_dirs = ['src'],
+			 libraries = ['m'],
                          depends = ['src/interpolate.h'])
 
     config.add_data_dir('tests')
--- scipy-0.12.0/scipy/ndimage/setup.py.orig	2013-07-20 21:12:11.681281002 -0300
+++ scipy-0.12.0/scipy/ndimage/setup.py	2013-07-20 21:15:14.003289325 -0300
@@ -13,6 +13,7 @@ def configuration(parent_package='', top
                  "src/ni_fourier.c","src/ni_interpolation.c",
                  "src/ni_measure.c",
                  "src/ni_morphology.c","src/ni_support.c"],
+	libraries = ['m'],
         include_dirs=['src']+[get_include()],
     )
 
--- scipy-0.12.0/scipy/optimize/setup.py.orig	2013-07-20 21:12:19.880281376 -0300
+++ scipy-0.12.0/scipy/optimize/setup.py	2013-07-20 21:15:55.266291209 -0300
@@ -16,7 +16,8 @@ def configuration(parent_package='',top_
 
     config.add_library('rootfind',
                        sources=[join('Zeros','*.c')],
-                       headers=[join('Zeros','zeros.h')])
+                       headers=[join('Zeros','zeros.h')],
+    		       libraries=['m'])
 
     config.add_extension('_zeros',
                          sources=['zeros.c'],
@@ -31,7 +32,8 @@ def configuration(parent_package='',top_
     sources=['moduleTNC.c','tnc.c']
     config.add_extension('moduleTNC',
                          sources=[join('tnc',x) for x in sources],
-                         depends=[join('tnc','tnc.h')])
+                         depends=[join('tnc','tnc.h')],
+			 libraries=['m'])
 
     config.add_extension('_cobyla',
                          sources=[join('cobyla',x) for x in ['cobyla.pyf',
--- scipy-0.12.0/scipy/signal/setup.py.orig	2013-07-20 21:12:37.297282171 -0300
+++ scipy-0.12.0/scipy/signal/setup.py	2013-07-20 21:17:05.458294413 -0300
@@ -14,14 +14,16 @@ def configuration(parent_package='', top
                                   'medianfilter.c', 'lfilter.c.src',
                                   'correlate_nd.c.src'],
                          depends=['sigtools.h'],
-                         include_dirs=['.']
+                         include_dirs=['.'],
+			 libraries = ['m']
     )
 
-    config.add_extension('_spectral', sources=['_spectral.c'])
+    config.add_extension('_spectral', sources=['_spectral.c'], libraries = ['m'])
 
     config.add_extension('spline',
         sources=['splinemodule.c', 'S_bspline_util.c', 'D_bspline_util.c',
                  'C_bspline_util.c', 'Z_bspline_util.c', 'bspline_util.c'],
+	libraries = ['m']
     )
 
     return config
--- scipy-0.12.0/scipy/spatial/setup.py.orig	2013-07-20 21:12:49.453282726 -0300
+++ scipy-0.12.0/scipy/spatial/setup.py	2013-07-20 21:17:55.970296719 -0300
@@ -32,10 +32,11 @@ def configuration(parent_package = '', t
                          sources=['qhull.c'] + qhull_src,
                          **cfg)
 
-    config.add_extension('ckdtree', sources=['ckdtree.c']) # FIXME: cython
+    config.add_extension('ckdtree', sources=['ckdtree.c'], libraries = ['m']) # FIXME: cython
 
     config.add_extension('_distance_wrap',
         sources=[join('src', 'distance_wrap.c'), join('src', 'distance.c')],
+	libraries = ['m'],
         include_dirs = [get_numpy_include_dirs()])
 
     return config
--- scipy-0.12.0/scipy/stats/setup.py.orig	2013-07-20 21:13:00.110283213 -0300
+++ scipy-0.12.0/scipy/stats/setup.py	2013-07-20 21:18:22.767297942 -0300
@@ -23,6 +23,7 @@ def configuration(parent_package='',top_
     # add vonmises_cython module
     config.add_extension('vonmises_cython',
         sources=['vonmises_cython.c'], # FIXME: use cython source
+	libraries = ['m']
     )
 
     # add _rank module
