diff --git a/scipy/cluster/setup.py b/scipy/cluster/setup.py
index 715bbac..2e9a63f 100755
--- a/scipy/cluster/setup.py
+++ b/scipy/cluster/setup.py
@@ -17,11 +17,13 @@ def configuration(parent_package = '', top_path = None):
     config.add_extension('_vq',
         sources=[join('src', 'vq_module.c'), join('src', 'vq.c')],
         include_dirs = [get_numpy_include_dirs()],
+	libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     config.add_extension('_hierarchy_wrap',
         sources=[join('src', 'hierarchy_wrap.c'), join('src', 'hierarchy.c')],
         include_dirs = [get_numpy_include_dirs()],
+	libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     return config
diff --git a/scipy/interpolate/setup.py b/scipy/interpolate/setup.py
index 8b36042..4f440c9 100755
--- a/scipy/interpolate/setup.py
+++ b/scipy/interpolate/setup.py
@@ -12,7 +12,8 @@ def configuration(parent_package='',top_path=None):
                       )
 
     config.add_extension('interpnd',
-                         sources=['interpnd.c'])
+                         sources=['interpnd.c'],
+    			 libraries=['m'])
 
     config.add_extension('_fitpack',
                          sources=['src/_fitpackmodule.c'],
@@ -28,6 +29,7 @@ def configuration(parent_package='',top_path=None):
     config.add_extension('_interpolate',
                          sources=['src/_interpolate.cpp'],
                          include_dirs = ['src'],
+			 libraries = ['m'],
                          depends = ['src/interpolate.h'])
 
     config.add_data_dir('tests')
diff --git a/scipy/ndimage/setup.py b/scipy/ndimage/setup.py
index 115769f..7217bd7 100644
--- a/scipy/ndimage/setup.py
+++ b/scipy/ndimage/setup.py
@@ -11,6 +11,7 @@ def configuration(parent_package='', top_path=None):
                  "src/ni_fourier.c","src/ni_interpolation.c",
                  "src/ni_measure.c",
                  "src/ni_morphology.c","src/ni_support.c"],
+	libraries = ['m'],
         include_dirs=['src']+[get_include()],
     )
 
diff --git a/scipy/optimize/setup.py b/scipy/optimize/setup.py
index 3beb5fb..a44e247 100755
--- a/scipy/optimize/setup.py
+++ b/scipy/optimize/setup.py
@@ -15,7 +15,8 @@ def configuration(parent_package='',top_path=None):
 
     config.add_library('rootfind',
                        sources=[join('Zeros','*.c')],
-                       headers=[join('Zeros','zeros.h')])
+                       headers=[join('Zeros','zeros.h')],
+    		       libraries=['m'])
 
     config.add_extension('_zeros',
                          sources=['zeros.c'],
@@ -30,7 +31,8 @@ def configuration(parent_package='',top_path=None):
     sources=['moduleTNC.c','tnc.c']
     config.add_extension('moduleTNC',
                          sources=[join('tnc',x) for x in sources],
-                         depends=[join('tnc','tnc.h')])
+                         depends=[join('tnc','tnc.h')],
+			 libraries=['m'])
 
     config.add_extension('_cobyla',
                          sources=[join('cobyla',x) for x in ['cobyla.pyf',
diff --git a/scipy/signal/setup.py b/scipy/signal/setup.py
index 34f84e3..2d03cdf 100755
--- a/scipy/signal/setup.py
+++ b/scipy/signal/setup.py
@@ -13,14 +13,16 @@ def configuration(parent_package='', top_path=None):
                                   'medianfilter.c', 'lfilter.c.src',
                                   'correlate_nd.c.src'],
                          depends=['sigtools.h'],
-                         include_dirs=['.']
+                         include_dirs=['.'],
+			 libraries = ['m']
     )
 
-    config.add_extension('spectral', sources=['spectral.c'])
+    config.add_extension('spectral', sources=['spectral.c'], libraries = ['m'])
 
     config.add_extension('spline',
         sources=['splinemodule.c', 'S_bspline_util.c', 'D_bspline_util.c',
                  'C_bspline_util.c', 'Z_bspline_util.c', 'bspline_util.c'],
+	libraries = ['m']
     )
 
     return config
diff --git a/scipy/spatial/setup.py b/scipy/spatial/setup.py
index 7401d55..d5eba04 100755
--- a/scipy/spatial/setup.py
+++ b/scipy/spatial/setup.py
@@ -34,10 +34,11 @@ def configuration(parent_package = '', top_path = None):
                          libraries=libs,
                          **lapack)
 
-    config.add_extension('ckdtree', sources=['ckdtree.c']) # FIXME: cython
+    config.add_extension('ckdtree', sources=['ckdtree.c'], libraries = ['m']) # FIXME: cython
 
     config.add_extension('_distance_wrap',
         sources=[join('src', 'distance_wrap.c'), join('src', 'distance.c')],
+	libraries = ['m'],
         include_dirs = [get_numpy_include_dirs()])
 
     return config
diff --git a/scipy/stats/setup.py b/scipy/stats/setup.py
index 1ada8be..a401457 100755
--- a/scipy/stats/setup.py
+++ b/scipy/stats/setup.py
@@ -22,6 +22,7 @@ def configuration(parent_package='',top_path=None):
     # add vonmises_cython module
     config.add_extension('vonmises_cython',
         sources=['vonmises_cython.c'], # FIXME: use cython source
+	libraries = ['m']
     )
 
     # add _rank module
